src/models/comment.dao.ts

```diff
@@ -1,54 +1,160 @@
 import dataSource from './database';
 
-async function addCommentOnPost(userId: string, postId: string, comment: string) {
-  return await dataSource.query(`
-    INSERT INTO comments(
-      comment,
-      posting_id,
-      user_id
-      )
-    VALUES(?, ?, ?);`
-  , [comment, postId, userId])
-}
-
-async function findCommentById(commentId: string) {
-  const [comment] = await dataSource.query(`
-      SELECT
-        comments.comment,
-        postings.id AS post_id,
-        comments.created_at,
-        comments.updated_at,
-        users.nickname as user_nickname,
-        users.id as user_id
-      FROM comments
-      JOIN users    ON comments.user_id     = users.id
-      JOIN postings ON comments.posting_id  = postings.id
-      WHERE comments.id = ?`
-      , [commentId]
-    );
-  return comment as {comment: string, poast_id: string, created_at: string, updated_at: string, user_nickname: string, user_id: string, };
-}
-
-async function updateComment(commentId: string, comment: string) {
+const addCommentOnPost = async (userId: number, postId: number, comment: string, is_secret: number) => {
   await dataSource.query(`
-      UPDATE comments
-        SET comment = ?
-      WHERE id = ?`
-      , [comment, commentId]
+    INSERT INTO comments (
+      users_id,
+      company_posts_id,
+      comment_content,
+      depth,
+      sequence,
+      is_secret
     )
+    VALUES (?, ?, ?, ?, ?, ?)`
+  , [userId, postId, comment, 1, 1, is_secret])
 }
 
-async function deleteComment(commentId: string) {
+const addCommentOnComment = async (userId: number, postId: number, commentId: number, comment: string, SEQ: number, is_secret: number) => {
+  await dataSource.query(`
+    INSERT INTO comments (
+      users_id,
+      company_posts_id,
+      comment_content,
+      comments_id,
+      depth,
+      sequence,
+      is_secret
+    )
+    VALUES (?, ?, ?, ?, ?, ?, ?)`
+  , [userId, postId, comment, commentId, 2, SEQ + 1, is_secret])
+}
+
+const getCommentOnPost = async (postId: number, pagenation: number): Promise<any> => {
+  const result = await dataSource.query(`
+    SELECT
+      cmt.id,
+      cmt.users_id,
+      users.username,
+      cmt.comment_content,
+      cmt.comments_id,
+      cmt.depth,
+      cmt.sequence,
+      cmt.created_at,
+      cmt.is_secret
+    FROM
+      comments as cmt
+    JOIN company_posts as cp ON cp.id = cmt.company_posts_id
+    JOIN users ON users.id = cmt.users_id
+    WHERE cmt.company_posts_id = ?
+    ORDER BY IF(ISNULL(cmt.comments_id), cmt.id, cmt.comments_id), sequence
+    LIMIT ${pagenation}, 20
+  `, [postId])
+  return result as CommentType;
+}
```

Comment:
기획에 댓글 20개씩이라고 적혀있지만, 백엔드에서는 개수를 유동적으로 조정할 수 있게끔 개발해두시는 것이 좋습니다.
dataLength를 인자로 받아오되, 함수의 인자로 받으면서 default 값을 20으로 지정해두시는 것이 좋겠습니다.

```suggestion
const getCommentOnPost = async (postId: number, pagenation: number, dataLength?:number = 20): Promise<any> => {
  const result = await dataSource.query(`
    SELECT
      cmt.id,
      cmt.users_id,
      users.username,
      cmt.comment_content,
      cmt.comments_id,
      cmt.depth,
      cmt.sequence,
      cmt.created_at,
      cmt.is_secret
    FROM
      comments as cmt
    JOIN company_posts as cp ON cp.id = cmt.company_posts_id
    JOIN users ON users.id = cmt.users_id
    WHERE cmt.company_posts_id = ?
    ORDER BY IF(ISNULL(cmt.comments_id), cmt.id, cmt.comments_id), sequence
    LIMIT ${pagenation}, ${dataLength}
  `, [postId])
  return result as CommentType;
}
```

Reply:
반영했습니다. 감사합니다~!